app/Models/User.php

```diff
@@ -1042,6 +1045,17 @@ public function isFullBN(?string $mode = null)
         return $this->inGroupWithPlaymode('bng', $mode);
     }
 
+    public function isForumModerator(Forum\Forum $forum): bool
+    {
+        // permission set directly though moderator_groups should behave like checks
+        // using findUserGroup directly instead of isGroup and are available with OAuth.
+        $token = $this->token();
+
+        return $token !== null && !($token->delegatesOwner() && $token->can('forum.delegate')) && !$token->isOwnToken()
```

Comment:
the chat announce is easier to control (as those involve chat scopes) but auth code for own client giving out forum moderator permission on public scope (reading deleted posts for example) is probably not an expected outcome (and it's currently already problematic in even worse way).

which is why I suggested adding scope which simply enable all user group access instead of adding more specific exceptions on specific places (which potentially gives more permission than expected)

Reply:
come to think of it, the whole own token thing was for the whole applying to run a chat bot rules thing :|